From e470df7d72c2befc04b2b5eeb07a3a9d3c4532dd Mon Sep 17 00:00:00 2001
From: shreyacsingh <shreyacsingh@google.com>
Date: Mon, 27 Mar 2023 17:01:06 +0000
Subject: [PATCH] Remove factory-reset logic from post-suw syncauth

Bug: 266906483 , Bug: 278691965
Test: Manual
Dialog-before: https://drive.google.com/file/d/1uE02duXkFGG1G6qu4foNgGKGLgQhIWoO/view?usp=sharing
Dialog-after: https://drive.google.com/file/d/1UM7w-ClnnhWi3aVokhXrZk42e8gdbQSI/view?usp=sharing
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:b16f50a817da9e48d0f3c572de832093362b46e2)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:6d51ea62287e8c6c55bf54c9913bb6d982f7e64e)
Merged-In: I0da6a2eaad20f628973ca3a080602992dfaeb63e
Change-Id: I0da6a2eaad20f628973ca3a080602992dfaeb63e
---
 .../provisioning/AdminIntegratedFlowPrepareActivity.java   | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/src/com/android/managedprovisioning/provisioning/AdminIntegratedFlowPrepareActivity.java b/src/com/android/managedprovisioning/provisioning/AdminIntegratedFlowPrepareActivity.java
index d9245654..cf195802 100644
--- a/src/com/android/managedprovisioning/provisioning/AdminIntegratedFlowPrepareActivity.java
+++ b/src/com/android/managedprovisioning/provisioning/AdminIntegratedFlowPrepareActivity.java
@@ -101,7 +101,12 @@ public class AdminIntegratedFlowPrepareActivity extends AbstractProvisioningActi
 
     @Override
     protected void decideCancelProvisioningDialog() {
-        showCancelProvisioningDialog(/* resetRequired = */true);
+        if (getUtils().isDeviceOwnerAction(mParams.provisioningAction)
+                || getUtils().isOrganizationOwnedAllowed(mParams)) {
+            showCancelProvisioningDialog(/* resetRequired= */ true);
+        } else {
+            showCancelProvisioningDialog(/* resetRequired= */ false);
+        }
     }
 
     private void initializeUi(ProvisioningParams params) {
-- 
2.41.0.255.g8b1d071c50-goog

